File: api/app/models/oilseed.py
from datetime import date, datetime

from sqlalchemy import Date, DateTime, Enum, ForeignKey, func
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import Base
from app.models.enums import OilseedCrop


class Oilseed(Base):
    __tablename__ = "oilseeds"

    id: Mapped[int] = mapped_column(primary_key=True)
    land_id: Mapped[int] = mapped_column(ForeignKey("lands.id"), index=True)
    crop: Mapped[OilseedCrop] = mapped_column(Enum(OilseedCrop, name="oilseed_planting_crop"))
    sowing_date: Mapped[date] = mapped_column(Date)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now())

File: tilahan_saathi_nit/lib/models/oilseed.dart
import 'package:tilahan_saathi/models/enums.dart';

class Oilseed {
  const Oilseed({
    required this.id,
    required this.landId,
    required this.crop,
    required this.sowingDate,
    required this.createdAt,
  });

  factory Oilseed.fromJson(Map<String, dynamic> json) {
    return Oilseed(
      id: json['id'] as int,
      landId: json['land_id'] as int,
      crop: OilseedCrop.fromApi(json['crop'] as String),
      sowingDate: DateTime.parse(json['sowing_date'] as String),
      createdAt: DateTime.parse(json['created_at'] as String),
    );
  }

  final int id;
  final int landId;
  final OilseedCrop crop;
  final DateTime sowingDate;
  final DateTime createdAt;
}
